pyboy/utils.py

```diff
@@ -273,3 +274,146 @@ def __init__(
         self.mouse_scroll_x = mouse_scroll_x
         self.mouse_scroll_y = mouse_scroll_y
         self.mouse_button = mouse_button
+
+
+##############################################################
+# Memory Scanning
+#
+class EndianType(Enum):
+    """Enumeration for defining endian types."""
+    LITTLE = 1
+    BIG = 2
+
+class CompareType(Enum):
+    """Enumeration for defining types of comparisons."""
+    EXACT = 1
+    LESS_THAN = 2
+    GREATER_THAN = 3
+    LESS_THAN_OR_EQUAL = 4
+    GREATER_THAN_OR_EQUAL = 5
+
+class ScanMode(Enum):
+    """Enumeration for defining scanning modes."""
+    INT = 1
+    BCD = 2
+
+class BCDConverter:
+    def dec_to_bcd(self, value, byte_width=1, endian_type=EndianType.LITTLE):
+        """
+        Converts a decimal value to Binary Coded Decimal (BCD).
+
+        :param value: Integer value to convert.
+        :param byte_width: The number of bytes to consider for each value.
+        :param endian_type: The endian type to use. Note, this is only used for 16-bit values and higher.
+        :return: BCD equivalent of the decimal value.
+        """
+        bcd_result = []
+        for _ in range(byte_width):
+            tens = ((value%100) // 10) << 4
+            units = value % 10
+            bcd_byte = (tens | units) & 0xFF
+            bcd_result.append(bcd_byte)
+            value //= 100
+        if endian_type == EndianType.BIG:
+            return int.from_bytes(bcd_result, byteorder='big')
+        else:
+            return int.from_bytes(bcd_result, byteorder='little')
+
+    def bcd_to_dec(self, value, byte_width=1, endian_type=EndianType.LITTLE):
+        """
+        Converts a Binary Coded Decimal (BCD) value to decimal.
+
+        :param value: BCD value to convert.
+        :param byte_width: The number of bytes to consider for each value.
+        :param endian_type: The endian type to use. Note, this is only used for 16-bit values and higher.
+        :return: Decimal equivalent of the BCD value.
+        """
+        decimal_value = 0
+        self._multiplier = 1
+
+        bcd_bytes = value.to_bytes(byte_width, 'big' if endian_type == EndianType.BIG else 'little')
+        
+        for bcd_byte in bcd_bytes:
+            decimal_value += ((bcd_byte >> 4) * 10 + (bcd_byte & 0x0F)) * self._multiplier
+            self._multiplier *= 100
+
+        return decimal_value
+
+class MemoryScanner():
```

Comment:
Rebase complete.

Reply:
Actually that commit is there because its on master and not v2.0.0, so as long as you are okay with that commit being in this pull i will leave it be.

Reply:
Fantastic, I'll verify the history before merging your PR